apis/core/v1beta1/flagd_types.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// FlagdSpec defines the desired state of Flagd
+type FlagdSpec struct {
+	// Replicas defines the number of replicas to create for the service
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// ServiceType represents the type of Service to create.
+	// Must be one of: ClusterIP, NodePort, LoadBalancer, and ExternalName.
+	// Default: ClusterIP
+	// +optional
+	// +kubebuilder:default=ClusterIP
+	// +kubebuilder:validation:Enum:=ClusterIP;NodePort;LoadBalancer;ExternalName
+	ServiceType v1.ServiceType `json:"serviceType,omitempty"`
+
+	// ServiceAccountName the service account name for the flagd deployment
+	// +optional
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
+
+	// OtelCollectorUri defines the OpenTelemetry collector URI to enable OpenTelemetry Tracing in flagd.
+	// +optional
+	OtelCollectorUri string `json:"otelCollectorUri"`
```

Comment:
This already [exists](https://github.com/open-feature/open-feature-operator/blob/b58261976d03059fe7b214244eb8087519cbbc7c/apis/core/v1beta1/featureflagsource_types.go#L105-L107) on the `FeatureFlagSource` spec. Do the do the same thing? I'm a bit confused why it's on both :sweat_smile:

Reply:
I think this is not relevant to the service definition. Same goes for the `FeatureFlagSource` 🤔

Reply:
@toddbaert you are right, having this here is not required, as it's already in the FeatureFlagSource - will remove this